plugins/packages/zendesk/lib/index.ts

```diff
@@ -0,0 +1,171 @@
+        headers: { 'Content-Type': 'application/json' },
+      });
+
+      const result = JSON.parse(response.body);
+
+      if (response.statusCode !== 200) {
+        throw Error('could not connect to Zendesk');
+      }
+
+      if (result['access_token'] && result['token_type'] === 'bearer') {
+        authDetails.push(['access_token', result['access_token']]);
+      }
+
+      if (result['refresh_token']) {
+        authDetails.push(['refresh_token', result['refresh_token']]);
+      }
+    } catch (error) {
+      console.log(error.response.body);
+      throw Error('could not connect to Zendesk');
+    }
+
+    return authDetails;
+  }
+
+  async run(sourceOptions: SourceOptions, queryOptions: QueryOptions, dataSourceId: string): Promise<QueryResult> {
+    let result = {};
+    let response = null;
+    const operation = queryOptions.operation;
+    const subdomain = sourceOptions['subdomain'];
+    const accessToken = sourceOptions['access_token'];
+    const url = `https://${subdomain}.zendesk.com/api/v2`;
+
+    try {
+      switch (operation) {
+        case 'read_tickets':
+          response = await got(`${url}/tickets.json`, {
+            method: 'get',
+            headers: this.authHeader(accessToken),
+          });
+          result = JSON.parse(response.body);
+          break;
+
+        case 'list_users':
+          response = await got(`${url}/users`, {
+            method: 'get',
+            headers: this.authHeader(accessToken),
+          });
+          result = JSON.parse(response.body);
+          break;
+
+        case 'read_requested_tickets':
+          response = await got(`${url}/users/${queryOptions['user_id']}/tickets/requested`, {
+            method: 'get',
+            headers: this.authHeader(accessToken),
+          });
+          result = JSON.parse(response.body);
+          break;
+
+        case 'show_ticket':
+          response = await got(`${url}/tickets/${queryOptions['ticket_id']}`, {
+            method: 'get',
+            headers: this.authHeader(accessToken),
+          });
+          result = JSON.parse(response.body);
+          break;
+
+        case 'get_user':
+          response = await got(`${url}/users/${queryOptions['user_id']}`, {
+            method: 'get',
+            headers: this.authHeader(accessToken),
+          });
+          result = JSON.parse(response.body);
+          break;
+
+        case 'get_profile':
+          response = await got(`${url}/users/${queryOptions['user_id']}/profiles`, {
+            method: 'get',
+            headers: this.authHeader(accessToken),
+          });
+          result = JSON.parse(response.body);
+          break;
+
+        case 'update_ticket':
+          response = await got(`${url}/tickets/${queryOptions['ticket_id']}.json`, {
+            method: 'put',
+            headers: this.authHeader(accessToken),
+            body: JSON.stringify(queryOptions['body']),
+          });
+          result = JSON.parse(response.body);
```

Comment:
```suggestion
          result = await got(`${url}/tickets/${queryOptions['ticket_id']}.json`, {
            method: 'put',
            headers: this.authHeader(accessToken),
            json: queryOptions['body'],
          }).json();
```
`queryOptions['body']` is `string` or `json`? should parse if its `string`